test/jest/cookieSyncManager.spec.ts

```diff
@@ -0,0 +1,454 @@
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                { values: ['test'] },
+                true,
+                true, 
+            );
+        });
+
+        it('should update the urlWithRedirect if a redirectUrl is defined', () => {
+            const myPixelSettings: PixelConfiguration = {
+                pixelUrl: 'https://test.com',
+                redirectUrl: '?redirect=https://redirect.com&mpid=%%mpid%%',
+            };
+
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [{...pixelSettings, ...myPixelSettings}],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                        csd: {}
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync(null, testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                'https://test.com%3Fredirect%3Dhttps%3A%2F%2Fredirect.com%26mpid%3DtestMPID',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        // QUESTION: What is the purpose of this code path?
+        it('should call performCookieSync with mpid if previousMpid and mpid do not match', () => {
+            const mockSDK = ({
+                _Store: {
+                    webviewBridgeEnabled: false,
+                    pixelConfigurations: [pixelSettings],
+                },
+                _Persistence: {
+                    getPersistence: () => ({testMPID: {
+                    }}),
+                },
+            } as unknown) as MParticleWebSDK;
+
+            const cookieSyncManager = new CookieSyncManager(mockSDK);
+            cookieSyncManager.performCookieSync = jest.fn();
+
+            cookieSyncManager.attemptCookieSync('other-mpid', testMPID, true);
+
+            expect(cookieSyncManager.performCookieSync).toHaveBeenCalledWith(
+                '',
+                5,
+                testMPID,
+                {},
+                undefined,
+                true,
+                false, 
+            );
+        });
+
+        it('should call performCookieSync with mpid and csd if previousMpid and mpid do not match', () => {
```

Comment:
This test name and the previous test name make it sound like the same test.  but this one has stuff in CSD.  This one is probably more realistic because testMPID1 will have a CSD, before logging in to a new MPID.  So I think you can remove the previous test.

Reply:
I think both tests are valid because they each test two different code paths in what is a somewhat squirrely and complex function. I'd rather change the assertion to make them more distinct.